source/classes/Westley.js

```diff
@@ -58,24 +41,42 @@
 			throw new Error("Invalid command");
 		}
 		var commandComponents = commandTools.extractCommandComponents(command),
-			commandKey = commandComponents.shift();//,
-			//commandFilename = "command." + commandKey + ".js",
-			//commandExecute;
-		if (!availableCommands.hasOwnProperty(commandKey)) {
-			throw new Error("Unrecognised command: " + commandKey);
-		}
-		// try {
-		// 	commandExecute = require("__buttercup/classes/commands/" + commandFilename);
-		// } catch (err) {
-		// 	throw new Error("Unrecognised command: " + commandKey);
-		// }
-		var commandToExecute = availableCommands[commandKey];
+			commandKey = commandComponents.shift();
+
+		var commandObject = this._getCommandForKey(commandKey);
 		this._history.push(command);
-		commandToExecute.apply(commandToExecute, [this._dataset].concat(commandComponents));
+		commandObject.execute.apply(commandObject, [this._dataset].concat(commandComponents));
 		return this;
 	};
 
 	/**
+	 * Gets a command by its key from the cache with its dependencies injected
+	 * @param {String} commandKey The key of the command
+	 * @returns {Command} Returns the command
+	 * @memberof Westley
+	 */
+	Westley.prototype._getCommandForKey = function(commandKey) {
+		// If the command doesn't exist in the cache
+		if (this._cachedCommands[commandKey] === undefined) {
+			// Get the command object and inject its dependencies
+			var requirement = new (require("__buttercup/classes/commands/command." + commandKey + ".js"))();
+
+			if (requirement.injectSearching !== undefined) {
+				requirement.injectSearching(searching);
+			}
+
+			if (requirement.injectEntry !== undefined) {
+				requirement.injectEntry(entry);
+			}
```

Comment:
I agree with this - haven't looked over SOLID in several years though. I don't like how messy this makes the tiny command files, but when we go full ES6 maybe classes might even be an option here with the DI. Let's go with this then.